components/global/mixer-adapter/cmd/mixer-adapter/main.go

```diff
@@ -14,35 +14,76 @@
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
  * under the License.
+ *
  */
 
 package main
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
 	"os"
 
-	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/logging"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/logging"
 	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/wso2spadapter"
 )
 
-const defaultAdapterPort string = "38355"
+const (
+	defaultAdapterPort         string = "38355"
+	grpcAdapterCredentialPath  string = "GRPC_ADAPTER_CREDENTIAL"
+	grpcAdapterPrivateKeyPath  string = "GRPC_ADAPTER_PRIVATE_KEY"
+	grpcAdapterCertificatePath string = "GRPC_ADAPTER_CERTIFICATE"
+	spServerUrl                string = "SP_SERVER_URL"
+)
 
 func main() {
 
-	addr := defaultAdapterPort //Pre defined port for the adaptor. ToDo: Should get this as an environment variable
+	port := defaultAdapterPort //Pre defined port for the adaptor. ToDo: Should get this as an environment variable
 
 	if len(os.Args) > 1 {
-		addr = os.Args[1]
+		port = os.Args[1]
 	}
 
 	logger, err := logging.NewLogger()
 	if err != nil {
-		logger.Fatalf("Error building logger: ", err.Error())
+		log.Fatalf("Error building logger: %s", err.Error())
+	}
+	defer func() {
+		err := logger.Sync()
+		if err != nil {
+			log.Fatalf("Error syncing logger: %s", err.Error())
+		}
+	}()
+
+	/* Mutual TLS feature to secure connection between workloads
+	   This is optional. */
+	credential := os.Getenv(grpcAdapterCredentialPath)   // adapter.crt //change the name
+	privateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
+	certificate := os.Getenv(grpcAdapterCertificatePath) // ca.pem
+	spServerUrl := os.Getenv(spServerUrl)
+
+	logger.Infof("Sp server url : %s", spServerUrl)
+
+	client := &http.Client{}
+	publisher := wso2spadapter.SPMetricsPublisher{}
```

Comment:
Do we need SP in the struct? can't we use MetricsPublisher{}? Its anyway in the wso2spadapter package.